pkg/api/helpers.go

```diff
@@ -179,3 +180,65 @@ func LoadBalancerStatusDeepCopy(lb *LoadBalancerStatus) *LoadBalancerStatus {
 	}
 	return c
 }
+
+// GetAccessModesAsString returns a string representation of an array of access modes.
+// modes, when present, are always in the same order: RWO,ROX,RWX.
+func GetAccessModesAsString(modes []PersistentVolumeAccessMode) string {
+	modes = removeDuplicateAccessModes(modes)
+	modesAsString := ""
+
+	if containsAccessMode(modes, ReadWriteOnce) {
+		appendAccessMode(&modesAsString, "RWO")
+	}
+	if containsAccessMode(modes, ReadOnlyMany) {
+		appendAccessMode(&modesAsString, "ROX")
+	}
+	if containsAccessMode(modes, ReadWriteMany) {
+		appendAccessMode(&modesAsString, "RWX")
+	}
+
+	return modesAsString
+}
+
+// GetAccessModesAsString returns an array of AccessModes from a string created by GetAccessModesAsString
+func GetAccessModesFromString(modes string) []PersistentVolumeAccessMode {
+	strmodes := strings.Split(modes, ",")
+	accessModes := []PersistentVolumeAccessMode{}
+	for _, s := range strmodes {
+		if s == "RWO" {
+			accessModes = append(accessModes, ReadWriteOnce)
+		} else if s == "ROX" {
+			accessModes = append(accessModes, ReadOnlyMany)
+		} else if s == "RWX" {
+			accessModes = append(accessModes, ReadWriteMany)
+		}
+	}
+	return accessModes
+}
+
+func appendAccessMode(modes *string, mode string) {
+	if *modes != "" {
+		*modes += ","
+	}
+	*modes += mode
+}
+
+// removeDuplicateAccessModes returns an array of access modes without any duplicates
+func removeDuplicateAccessModes(modes []PersistentVolumeAccessMode) []PersistentVolumeAccessMode {
+	accessModes := []PersistentVolumeAccessMode{}
```

Comment:
Can't because of typing.  I would need a loop anyway.

`cannot use modes (type []PersistentVolumeAccessMode) as type []string in argument to util.NewStringSet`